### ISSUE TITLE
How to interact with several gate controllers?

### ISSUE THREAD

**Issue by moazam1** [uhppoted/uhppoted-codegen#11]
Hi - Just checking if there is any possibility to connect several gate controllers located in different locations/regions with a single server? The server URL will be same for all the devices. Can we flash firmware, etc. of the device so that the device can talk to the server and registered in our database when connected first time?

**Comment by uhppoted** [comment]
Yes, it's straightforward and there's no need to flash firmware or do anything special because the controller protocol is _request-response_ (i.e. the controller responds to the server that sent the request). 

However, to receiving events from the controllers the controllers have to be configured to send the events to a specific server, using a _set-event-listener_ command (e.g. [CLI set-listener](https://github.com/uhppoted/uhppote-cli#set-listener)).

One thing to note - the protocol as currently implemented in the _uhppoted_ libraries uses UDP and by default requests are sent as a UDP broadcast, which means the server and controllers have to be on the same LAN segment. For controllers and servers on different LAN segments, it is necessary to configure the IP addresses of the controllers in the _uhppoted.conf_ configuration file [CONTROLLERS](https://github.com/uhppoted/uhppoted/blob/master/documentation/uhppoted.conf/uhppoted.conf.md#controllers) section.

**Comment by uhppoted** [comment]
(BTW, am going to move this to the _uhppoted_ umbrella project _Discussions_ area)
